packages/full-stack-tests/src/hierarchy-builder/grouping/LabelGrouping.test.ts

```diff
@@ -0,0 +1,360 @@
+                    instanceKeys: [keys.childSubject1],
+                    children: false,
+                  }),
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject3],
+                    children: false,
+                  }),
+                ],
+              }),
+              NodeValidators.createForInstanceNode({
+                instanceKeys: [keys.childSubject2],
+                children: false,
+              }),
+            ],
+          }),
+        ],
+      });
+    });
+
+    it("creates different groups for different labels", async function () {
+      const labelGroupName1 = "test1";
+      const labelGroupName2 = "test2";
+      const { imodel, ...keys } = await buildIModel(this, async (builder) => {
+        const rootSubject = { className: subjectClassName, id: IModel.rootSubjectId };
+        const childSubject1 = insertSubject({ builder, codeValue: "test subject 1", parentId: rootSubject.id, userLabel: labelGroupName1 });
+        const childSubject2 = insertSubject({ builder, codeValue: "test subject 2", parentId: rootSubject.id, userLabel: labelGroupName2 });
+        const childSubject3 = insertSubject({ builder, codeValue: "test subject 3", parentId: rootSubject.id, userLabel: labelGroupName1 });
+        const childSubject4 = insertSubject({ builder, codeValue: "test subject 4", parentId: rootSubject.id, userLabel: labelGroupName2 });
+        return { rootSubject, childSubject1, childSubject2, childSubject3, childSubject4 };
+      });
+
+      await validateHierarchy({
+        provider: createProvider({ imodel, hierarchy: basicHierarchy }),
+        expect: [
+          NodeValidators.createForInstanceNode({
+            instanceKeys: [keys.rootSubject],
+            children: [
+              NodeValidators.createForLabelGroupingNode({
+                label: labelGroupName1,
+                children: [
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject1],
+                    children: false,
+                  }),
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject3],
+                    children: false,
+                  }),
+                ],
+              }),
+              NodeValidators.createForLabelGroupingNode({
+                label: labelGroupName2,
+                children: [
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject2],
+                    children: false,
+                  }),
+                  NodeValidators.createForInstanceNode({
+                    instanceKeys: [keys.childSubject4],
+                    children: false,
+                  }),
+                ],
+              }),
+            ],
+          }),
+        ],
+      });
+    });
+
+    describe("Label and Class grouping", () => {
```

Comment:
I'd move this into a separate `MultiLevelGrouping.test.ts` file.

Reply:
Moved